make/scripts/update_copyright_year.sh

```diff
@@ -49,66 +51,74 @@ rm -f -r ${tmp}
 mkdir -p ${tmp}
 total=0
 
-# Default or supplied company name
-if [ "$3" != "" ] ; then
-  company="$3"
-fi
-
-# This year or supplied year
-if [ "$2" != "" ] ; then
-  year="$2"
-else
-  year=`date +%Y`
-fi
-
-# VCS select
-vcs="$1"
+usage="Usage: `basename "$0"` [-c company] [-y year] [-h|f]"
+Help()
+{
+  # Display Help
+  echo "Updates the Copyright year range in Git sources."
+  echo
+  echo "By default, the tool limits the processed changesets "
+  echo "to those in the current branch and the current year."
+  echo
+  echo "Note, cancelling the script will skip cleanup in /tmp."
+  echo
+  echo $usage
+  echo "options:"
+  echo "c     Specifies the company. Set to Oracle by default."
+  echo "y     Specifies the copyright year. Set to current year by default."
+  echo "h     Print this help."
+  echo "f     Updates all change sets in a full year."
+  echo
+}
 
-if [ -z "$vcs" ] ; then
-  git_found=false
-  hg_found=false
+full_year=false
 
-  [ -d "${this_script_dir}/../../.git" ] && git_found=true
-  [ -d "${this_script_dir}/../../.hg" ] && hg_found=true
+# Process options
+while getopts "c:fhy:" option; do
+  case $option in
+    c) # supplied company year
+      company=${OPTARG}
+      ;;
+    f) # update all change sets in a full year
+      full_year=true
+      ;;
+    h) # display help
+      Help
+      exit 0
+      ;;
+    y) # supplied company year
+      year=${OPTARG}
+      ;;
+    \?) # illegal option
+      echo "$usage"
+      exit 1
+      ;;
+  esac
+done
 
-  if [ "$git_found" == "true" ] && [ "$hg_found" == "false" ] ; then
-    vcs="git"
-  elif [ "$hg_found" == "true" ] && [ "$git_found" == "false" ] ; then
-    vcs="hg"
+# VCS check
+git_found=false
+[ -d "${this_script_dir}/../../.git" ] && git_found=true
+if [ "$git_found" != "true" ]; then
+  echo "Error: could not auto-detect git as the version control system"
```

Comment:
Pre-existing. The error message is confusing (I first tried your script from a different location). Better would be adding something like this:

"Please execute script from within make/scripts"